src/io/flutter/sdk/FlutterSdkUtil.java

```diff
@@ -285,23 +328,35 @@ public static String parseFlutterSdkPath(String packagesFileContent) {
       final String flutterPrefix = "flutter:";
       if (line.startsWith(flutterPrefix)) {
         final String urlString = line.substring(flutterPrefix.length());
-
-        if (urlString.startsWith("file:")) {
-          final Url url = Urls.parseEncoded(urlString);
-          if (url == null) {
-            continue;
-          }
-          final String path = url.getPath();
-          // go up three levels
-          final File file = new File(url.getPath());
-          return file.getParentFile().getParentFile().getParentFile().getPath();
+        final String path = extractSdkPathFromUri(urlString);
+        if (path == null) {
+          continue;
         }
+        return path;
       }
     }
 
     return null;
   }
 
+  private static String extractSdkPathFromUri(String urlString) {
+    if (urlString.startsWith("file:")) {
+      final Url url = Urls.parseEncoded(urlString);
+      if (url == null) {
+        return null;
+      }
+      final String path = url.getPath();
+      // go up three levels for .packages or two for .dart_tool/package_config.json
```

Comment:
We could also parameterize this method - the call site knows whether it's 2 deep or 3; either way would work.

Reply:
I agree that a parameter is better; changed.